zone/attack.cpp

```diff
@@ -5809,6 +5812,252 @@ const DamageTable &Mob::GetDamageTable() const
 	return which[level - 50];
 }
 
+int Mob::GetMobFixedOffenseSkill()
+{
+	// Due to new code using a combination of Offense and Weapon skill to determine hit, depending on the class
+	// and weapon wielded by a mob, the hit rate of an equal level mob could vary between 15% and 60%, which made
+	// many mobs far too easy.  This particular call replaces the class based Offense Skill with a fixed value
+	// equal to that of a Warrior of appropriate Level if UseMobFixedOffenseSkill flag is TRUE.
+
+	int level = std::max(1,static_cast<int>(GetLevel()));
+
+	// Current tables are flat above Level 60
+	if (level > 60) {
+		level = 60;
+	}
+
+	int FixedOffenseSkillTable[] = {
```

Comment:
A lot of this looks like it could be done with math, with most values being `(level * 5) + 5`, any reason to not do such and just have exceptions where this changes? Looks to stay this way consistently to level 40. Example of using math.
```cpp
int Mob::GetMobFixedOffenseSkill()
{
	// Due to new code using a combination of Offense and Weapon skill to determine hit, depending on the class
	// and weapon wielded by a mob, the hit rate of an equal level mob could vary between 15% and 60%, which made
	// many mobs far too easy.  This particular call replaces the class based Offense Skill with a fixed value
	// equal to that of a Warrior of appropriate Level if UseMobFixedOffenseSkill flag is TRUE.

	int level = EQ::ClampUpper(std::max(1, static_cast<int>(GetLevel())), 60);

	if (level <= 40) {
		return (level * 5) + 5;
	} else if (EQ::ValueWithin(level, 41, 50)) {
		return 210;
	} else if (EQ::ValueWithin(level, 51, 58)) {
		return 210 + ((level - 50) * 5);
	}

	return 252;
}
```